src/Storageless/Http/SessionMiddleware.php

```diff
@@ -173,16 +174,14 @@ public static function fromAsymmetricKeyDefaults(
      * @throws \InvalidArgumentException
      * @throws \OutOfBoundsException
      */
-    public function __invoke(Request $request, Response $response, callable $out = null) : Response
+    public function process(Request $request, DelegateInterface $delegate) : Response
```

Comment:
I agree, I like the callable we had before. It's still not final...

I think this is the reason that it's gone:

> A more subjective, but also important, concern is that existing double-pass middleware typically uses the callable type hint to refer to middleware. This makes strict typing impossible, as there is no assurance that the callable being passed implements a middleware signature, which reduces runtime safety.
https://github.com/http-interop/http-middleware#44-chosen-approach

/cc @weierophinney